### PR TITLE
gio: Make ListStore::new() take a type parameter

### PR DESCRIPTION
**Pull request by BrainBlasted**
Makes passing the type more idiomatic.

### CODE REVIEW

File: gio/src/list_store.rs [hunk 1]
```diff
@@ -7,6 +7,11 @@ use glib::{prelude::*, translate::*, Object};
 use crate::{prelude::*, ListModel, ListStore};
 
 impl ListStore {
+    #[doc(alias = "g_list_store_new")]
+    pub fn new<T: StaticType>() -> ListStore {
```
Comment (bilelmoussaoui) [inline gio/src/list_store.rs:11]:
```suggestion
    pub fn new<T: StaticType>() -> Self {
```

File: gio/Gir.toml [hunk 2]
```diff
@@ -926,6 +926,9 @@ manual_traits = ["ListModelExtManual"]
 name = "Gio.ListStore"
 status = "generate"
 generate_builder = true
+    [[object.function]]
+    name = "new"
+    manual = true
```
Comment (sdroege) [inline gio/Gir.toml:931]: Please add a comment explaining why

File: gio/src/list_store.rs [hunk 3]
```diff
@@ -7,6 +7,11 @@ use glib::{prelude::*, translate::*, Object};
 use crate::{prelude::*, ListModel, ListStore};
 
 impl ListStore {
+    #[doc(alias = "g_list_store_new")]
+    pub fn new<T: StaticType>() -> Self {
```
Comment (sdroege) [inline gio/src/list_store.rs:11]: Do we also need a `pub fn with_type(type_: Type) -> Self` ?

File: gio/src/list_store.rs [hunk 4]
```diff
@@ -7,6 +7,16 @@ use glib::{prelude::*, translate::*, Object};
...
+        unsafe { from_glib_full(ffi::g_list_store_new(T::static_type().into_glib())) }
```
Comment (bilelmoussaoui) [inline gio/src/list_store.rs:12]:
```suggestion
        Self::with_type(T::static_type())
```

**Review by bilelmoussaoui — APPROVED** [review]

**Review by sdroege — APPROVED** [review]

**Pull request merged**